resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala

```diff
@@ -438,6 +441,24 @@ private[spark] class ApplicationMaster(
     registerAM(sparkConf, rpcEnv, driverRef, sparkConf.getOption("spark.driver.appUIAddress"),
       securityMgr)
 
+    // If the credentials file config is present, we must periodically renew tokens. So create
```

Comment:
Hmm, I know I already said "LGTM", but I think these two paths could be consolidated into one.

Basically, move this code to the `run()` method before either driver or executor launcher is started; then there's a single place where the token renewer is started, and then both client and cluster mode start it the same way.